systemd/systemd.go

```diff
@@ -614,14 +623,45 @@ func (s *systemd) getUnitStatus(properties []string, unitNames []string) ([]*Uni
 				}
 			}
 			if len(missing) > 0 {
-				return nil, fmt.Errorf("cannot get unit %q status: missing %s in ‘systemctl show’ output", cur.UnitName, strings.Join(missing, ", "))
+				return nil, fmt.Errorf("cannot get unit %q status: missing %s in ‘systemctl show’ output", cur.Id, strings.Join(missing, ", "))
 			}
 			sts = append(sts, cur)
+
+			// If we received more replies than what we requested, bail.
 			if len(sts) > len(unitNames) {
-				break // wut
+				break
+			}
+
+			// The 'systemctl' command can return the status parameters for a
+			// number of units at the same time. The status output between
+			// units are separated with an empty line. Every parsed status
+			// is appended to the 'sts' array, so the n'th entry matches the
+			// n'th request, as ordered in 'unitNames'. The 'sts' array can
+			// therefore be used as an index to the original request.
+			requestIndex := len(sts) - 1
+
+			// Record which unit 'Name' request produced this status because
+			// if the request was made using an aliased name, we must be
+			// consistent and reply using the same alias. We will check the
+			// validity of the alias below.
+			cur.Name = unitNames[requestIndex]
```

Comment:
this bit should be done earlier before the append, so that Name is available

Reply:
Fixed. Moved the 'missing parameters code' down.